subworkflows/local/bgc.nf

```diff
@@ -0,0 +1,187 @@
+                                        fna: [ meta, fna ]
+                                        gff: [ gff ]
+                                    }
+
+            ANTISMASH_ANTISMASHLITE ( ch_antismash_input.fna, ch_antismash_databases, ch_antismash_directory, ch_antismash_input.gff )
+
+        } else if ( params.annotation_tool == 'prokka' ) {
+
+            ch_antismash_input = gbk.filter {
+                                        meta, files ->
+                                            if ( meta.longest_contig < params.bgc_antismash_sampleminlength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of --bgc_antismash_sampleminlength ${params.bgc_antismash_sampleminlength}. Antismash will not be run for sample: ${meta.id}."
+                                            meta.longest_contig >= params.bgc_antismash_sampleminlength
+                                    }
+
+            ANTISMASH_ANTISMASHLITE ( ch_antismash_input, ch_antismash_databases, ch_antismash_directory, [] )
+
+        } else if ( params.annotation_tool == 'bakta' ) {
+
+            ch_antismash_input = gbk.filter {
+                                        meta, files ->
+                                            if ( meta.longest_contig < params.bgc_antismash_sampleminlength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of --bgc_antismash_sampleminlength ${params.bgc_antismash_sampleminlength}. Antismash will not be run for sample: ${meta.id}."
+                                            meta.longest_contig >= params.bgc_antismash_sampleminlength
+                                    }
+
+            ANTISMASH_ANTISMASHLITE ( ch_antismash_input, ch_antismash_databases, ch_antismash_directory, [] )
+
+        }
+
+        ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITE.out.versions)
+        ch_antismashresults_for_combgc = ANTISMASH_ANTISMASHLITE.out.knownclusterblast_dir
+            .mix(ANTISMASH_ANTISMASHLITE.out.gbk_input)
+            .groupTuple()
+            .map{
+                meta, files ->
+                [meta, files.flatten()]
+            }
+        ch_bgcresults_for_combgc = ch_bgcresults_for_combgc.mix(ch_antismashresults_for_combgc)
+
+    }
+
+    // DEEPBGC
+    if ( !params.bgc_skip_deepbgc ){
+        if ( params.bgc_deepbgc_database ) {
+
+            ch_deepbgc_database = Channel
+                .fromPath( params.bgc_deepbgc_database )
+                .first()
+        } else {
+            DEEPBGC_DOWNLOAD()
+            ch_deepbgc_database = DEEPBGC_DOWNLOAD.out.db
+        }
+
+    ch_deepbgc_input = fna
+
+    DEEPBGC_PIPELINE ( ch_deepbgc_input, ch_deepbgc_database)
+    ch_versions = ch_versions.mix(DEEPBGC_PIPELINE.out.versions)
+    ch_bgcresults_for_combgc = ch_bgcresults_for_combgc.mix(DEEPBGC_PIPELINE.out.bgc_tsv)
+    }
+
+    // GECCO
+    if ( !params.bgc_skip_gecco ) {
+        ch_gecco_input = fna.groupTuple()
```

Comment:
Why do we have to groupTuple here?Can we apply all FASTAs for a single sample in one run, whereas the other tools do it one by one? That said, we only already expect ona fasta per sample already according to the input sheet